nozzle/api_client.go

```diff
@@ -96,55 +110,40 @@ func (api *APIClient) listApps() map[string]*AppInfo {
 	return appsInfo
 }
 
+func (api *APIClient) GetUncached(key string) (*AppInfo, error) {
+	app, err := api.client.AppByGuid(key)
+	if err != nil {
+		return nil, err
+	}
+	return newAppInfo(app), nil
+}
+
 // GetApp return cached AppInfo for a guid
 func (api *APIClient) GetApp(guid string) (*AppInfo, error) {
-	size := metrics.GetOrRegisterGauge("cache.size", nil)
-	errors := metrics.GetOrRegisterCounter("cache.errors", nil)
-	miss := metrics.GetOrRegisterCounter("cache.miss", nil)
-	size.Update(int64(api.appCache.ItemCount()))
+	//size := metrics.GetOrRegisterGauge("cache.size", nil)
```

Comment:
Can use a `NewFunctionalGauge` and supply a func that emits the cache size once per report interval rather than update this on every call to GetApp.